### PR TITLE
Avoid boolean blindness with choice

### PR DESCRIPTION
**Pull request by brandonchinn178**
Broken off from https://github.com/tweag/ormolu/pull/1097#discussion_r1491404015

Broke off some easy places. Not urgent

Pros:
- Prevents boolean blindness when constructing literals
- Prevents mixing up two boolean values

Cons:
- Casting to/from bool is a bit tedious. See https://github.com/mboes/choice/pull/6

### CODE REVIEW

**Review by amesgen — APPROVED** [review]
Looks great, thanks for following up on that comment :smile: 

There are additional places (eg replacing `IsApplicand`) that could be changed, but this can happen incrementally.

I like https://github.com/mboes/choice/pull/6, but I would also be fine to merge this now.

**Comment by brandonchinn178** [comment]
@amesgen Do you know how to update the Nix config to fetch the new `choice` version that was released to Hackage?

**Review by mrkkrp — APPROVED** [review]
Very nice! I left one minor comment.

**Comment by mrkkrp** [comment]
Can you also rebase?

**Comment by amesgen** [comment]
> @amesgen Do you know how to update the Nix config to fetch the new `choice` version that was released to Hackage?

Pushed a corresponding commit :+1: (`nix flake update haskellNix`)

**Comment by brandonchinn178** [comment]
@amesgen Thanks. FWIW I don't have Nix installed; it's a bit difficult to make that change to make that change as a non-Nix contributor...

File: ormolu.cabal [hunk 1]
```diff
@@ -175,6 +176,7 @@ test-suite tests
         Cabal-syntax >=3.12 && <3.13,
         QuickCheck >=2.14,
         base >=4.14 && <5,
+        choice >=0.1 && <0.3,
```
[the remark this answers is not in the document]
Comment (brandonchinn178) [reply ormolu.cabal]: Yes, that was an oversight. Thanks

**Pull request merged**